appinventor/appengine/src/com/google/appinventor/client/ConnectProgressBar.java

```diff
@@ -0,0 +1,89 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2019 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.client;
+
+import com.google.appinventor.client.explorer.dialogs.ProgressBarDialogBox;
+
+import com.google.appinventor.client.explorer.project.Project;
+
+import static com.google.appinventor.client.Ode.MESSAGES;
+
+/**
+ * Show a Progress Bar during connection from the Browser to the Companion.
+ * We are designed by be used from a static context. This facilitates calls
+ * from both the Java (GWT) side of things and the Javascript (blockly) side
+ * of thing.
+ *
+ * Javascript starts the progress bar when we learn the connection information
+ * we need from the Rendezvous server. After a connection to the Companion is complete
+ * we hand control over to the AssetManager (in GWT land).
+ *
+ * @author jis@mit.edu (Jeffrey I. Schiller)
+ *
+ */
+
+
+public final class ConnectProgressBar {
+
+    private static ConnectProgressBar INSTANCE = new ConnectProgressBar();
+    private Project currentProject;
+    private ProgressBarDialogBox progressBar;
+
+    private ConnectProgressBar() {
+      exportMethodsToJavascript();
+    }
+
+    public static ConnectProgressBar getInstance() {
```

Comment:
This is only called from Ode, and the result isn't used. We could just make this a void function with the name initialize() and it would be more semantically accurate.